### PR TITLE
Factor out digitize buttons

### PR DESCRIPTION
**Pull request by hwbllmnn**
This factores out the digitizing buttons (line, point, polygon) from the redlining component. The buttons can be used seperately and standalone with either a layer or a ol.Collection as a configuration option.

This PR also includes linting fixes (atom only sees the correct .eslintrc if you open BasiGX as a separate project).

@terrestris/devs please review

### CODE REVIEW

File: src/view/container/Redlining.js [hunk 1]
```diff
@@ -301,74 +283,17 @@ Ext.define('BasiGX.view.container.Redlining', {
...
-                        me.map.addInteraction(me.drawPointInteraction);
-                    }
-                    if (pressed) {
-                        me.drawPointInteraction.setActive(true);
-                    } else {
-                        me.drawPointInteraction.setActive(false);
-                    }
-                }
-            }
+            xtype: 'basigx-button-digitize-point',
```
Comment (annarieger) [inline src/view/container/Redlining.js]: Please add `BasiGX.view.button.DigitizePoint` to requires block

File: src/view/container/Redlining.js [hunk 2]
```diff
@@ -301,74 +283,17 @@ Ext.define('BasiGX.view.container.Redlining', {
...
-                        me.map.addInteraction(me.drawLineInteraction);
-                    }
-                    if (pressed) {
-                        me.drawLineInteraction.setActive(true);
-                    } else {
-                        me.drawLineInteraction.setActive(false);
-                    }
-                }
-            }
+            xtype: 'basigx-button-digitize-line',
```
Comment (annarieger) [inline src/view/container/Redlining.js]: Please add `BasiGX.view.button.DigitizeLine` to requires block

File: src/view/container/Redlining.js [hunk 3]
```diff
@@ -301,74 +283,17 @@ Ext.define('BasiGX.view.container.Redlining', {
...
-                    }
-                    if (pressed) {
-                        me.drawPolygonInteraction.setActive(true);
-                    } else {
-                        me.drawPolygonInteraction.setActive(false);
-                    }
-                }
-            }
+            xtype: 'basigx-button-digitize-polygon',
+            collection: me.redlineFeatures,
```
Comment (annarieger) [inline src/view/container/Redlining.js]: Please add `BasiGX.view.button.DigitizePolygon` to requires block

File: src/view/container/Redlining.js (hunk 1 above)
Comment (hwbllmnn) [reply src/view/container/Redlining.js]: Good catch, done.

File: src/view/container/Redlining.js (hunk 2 above)
Comment (hwbllmnn) [reply src/view/container/Redlining.js]: Good catch, done.

File: src/view/container/Redlining.js (hunk 3 above)
Comment (hwbllmnn) [reply src/view/container/Redlining.js]: Good catch, done.

**Comment by annarieger** [comment]
Thanks, please merge at will :pineapple:

**Pull request merged**